### PR TITLE
Fix AsyncResource propagation issue

### PR DESCRIPTION
**Pull request by hyperair**
Fixes: [#200](https://github.com/sindresorhus/p-queue/issues/200)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import {AsyncResource} from 'node:async_hooks';
```
Comment (sindresorhus) [inline index.js:1]: This package is used in the browser a lot too, so you cannot directly call Node.js only APIs.

You could create a small wrapper for asyncresource.bind, put it in a file, make another placeholder one that does nothing, and then use subpath imports to delegate the placeholder one to browser and the real one to Node.js: https://nodejs.org/api/packages.html#subpath-imports

File: test.js [hunk 2]
```diff
@@ -40,6 +41,23 @@ test('concurrency: 4', async t => {
...
+	const concurrency = 2;
+	const limit = pLimit(concurrency);
+	const store = new AsyncLocalStorage();
+
+	const checkId = async id => {
+		await Promise.resolve();
+		t.is(id, store.getStore()?.id);
+	};
+
+	const startCtx = async id => store.run({id}, () => limit(checkId, id));
```
Comment (sindresorhus) [inline test.js:54]:
```suggestion
	const startContext = async id => store.run({id}, () => limit(checkId, id));
```

File: index.js (hunk 1 above)
Comment (hyperair) [reply index.js]: Like so? https://github.com/sindresorhus/p-limit/pull/71/commits/9543ab0fc3e48e972a9e9bc5ad3928e21e16ed24

Comment (sindresorhus) [reply index.js]: Yes, exactly 👍

Comment (hyperair) [reply index.js]: Thanks!

**Pull request merged**